### DISCUSSION TITLE
Install git base to address BT doesn't compile

### DISCUSSION THREAD

**Discussion by lu7did** [earlephilhower/arduino-pico#1429] [General]
In response to my previous issue #1427 I'd been said the issue is fixed on the master but yet to be released, so in order to address it locally I should use the git version.

In order to do that I de-install the existing board manager (3.1.1) and installed it following the guidelines to install from git.

mkdir -p ~/Arduino/hardware/pico
git clone https://github.com/earlephilhower/arduino-pico.git ~/Arduino/hardware/pico/rp2040
cd ~/Arduino/hardware/pico/rp2040
git submodule update --init
cd pico-sdk
git submodule update --init
cd ../tools
python3 ./get.py

Still, when finished ok the board isn't available at the Arduino IDE as if it wasn't installed at all.

What else needs to be done for the Arduino IDE to recognize the Arduino core when installed from git?

Thanks in advance for your attention. Pedro.

**Comment by earlephilhower** [comment]
Did you install the IDE thru a Flatpack or a Snap, maybe?  They sometimes have issues with directory permissions and I recommend dumping them for the tarball install, if at all possible.

Can you also try a:
````
 ls -al ~/Arduino/hardware/pico/rp2040/
````

To verify things ended up where they need to be?

**Comment by lu7did** [comment]
Thank you for your answer Earle. This is a Mac, I installed the IDE from the official Arduino page, no idea what method they use for that.

The verification you recommended seems to show everything is ok.

PCOLLA@M ~ % ls -al ~/Arduino/hardware/pico/rp2040/
total 1992
drwxr-xr-x  27 PCOLLA  staff     864 May  5 12:35 .
drwxr-xr-x   3 PCOLLA  staff      96 May  5 12:34 ..
drwxr-xr-x  13 PCOLLA  staff     416 May  5 12:35 .git
drwxr-xr-x   3 PCOLLA  staff      96 May  5 12:35 .github
-rw-r--r--   1 PCOLLA  staff      89 May  5 12:35 .gitignore
-rw-r--r--   1 PCOLLA  staff    1596 May  5 12:35 .gitmodules
drwxr-xr-x   7 PCOLLA  staff     224 May  5 12:39 ArduinoCore-API
-rw-r--r--   1 PCOLLA  staff   26526 May  5 12:35 LICENSE
-rw-r--r--   1 PCOLLA  staff   14351 May  5 12:35 README.md
-rw-r--r--   1 PCOLLA  staff  937605 May  5 12:35 boards.txt
drwxr-xr-x  13 PCOLLA  staff     416 May  5 12:35 boot2
drwxr-xr-x   3 PCOLLA  staff      96 May  5 12:35 cores
drwxr-xr-x  46 PCOLLA  staff    1472 May  5 12:35 docs
drwxr-xr-x   7 PCOLLA  staff     224 May  5 12:35 include
-rw-r--r--   1 PCOLLA  staff    1184 May  5 12:35 keywords.txt
drwxr-xr-x  15 PCOLLA  staff     480 May  5 12:35 lib
drwxr-xr-x  50 PCOLLA  staff    1600 May  5 12:35 libraries
drwxr-xr-x  13 PCOLLA  staff     416 May  5 12:35 ota
drwxr-xr-x   8 PCOLLA  staff     256 May  5 12:35 package
-rw-r--r--   1 PCOLLA  staff     401 May  5 12:35 package.json
drwxr-xr-x  19 PCOLLA  staff     608 May  5 12:39 pico-sdk
-rw-r--r--   1 PCOLLA  staff   13767 May  5 12:35 platform.txt
-rw-r--r--   1 PCOLLA  staff     786 May  5 12:35 programmers.txt
drwxr-xr-x  10 PCOLLA  staff     320 May  5 12:40 system
drwxr-xr-x  11 PCOLLA  staff     352 May  5 12:35 tests
drwxr-xr-x  18 PCOLLA  staff     576 May  5 12:40 tools
drwxr-xr-x  61 PCOLLA  staff    1952 May  5 12:35 variants


Thanks in advance. Pedro.

**Reply by earlephilhower** [reply]
Ah, those instructions are for Linux.

On a Mac I have no idea where it should be, but it's probably not in `~/Arduino`.  You might want to look in your Documents folder (or whatever the Mac equivalent is), and if there's an `Arduino` folder there move the entire `hardware/...` tree into it.

**Reply by lu7did** [reply]
Yes, it installed now. Thank you very much!

**Reply by earlephilhower** [reply]
Great.  BTW, there is now a new official release anyway.  But now that you have the GIT version available, you can always update with a simple `git pull` and get the latest between releases.
